### DISCUSSION TITLE
Per-project configuration without reopening neovim

### DISCUSSION THREAD

**Discussion by ninofaivre** [VonHeikemen/lsp-zero.nvim#132] [Q&A] [answered]
Hi, I'm very happy with lsp and I successfully configured it to let's say have lua version to 5.4 in a project using this version and luaJIT for neovim directory or others projects using it. It work fine if I start neovim and go to one project, then close it and open it again and go to the other project. Is it possible to make it work without reopening neovim ?
Bellow a simple example of my config (sorry for my bad english)

initial config called :
```lua
local lsp = require('lsp-zero')

lsp.preset('per-project')

lsp.configure('sumneko_lua',
{
	settings =
	{
		Lua =
		{
			diagnostics =
			{
				globals = { 'vim', 'awesome' }
			}
		}
	}
})

lsp.setup()

vim.diagnostic.config({
	virtual_text = true,
})
```

first local project config file :
```lua
local lsp = require('lsp-zero')

lsp.use('sumneko_lua',
{
	settings =
	{
		Lua =
		{
			runtime =
			{
				version = 'LuaJIT'
			}
		}
	}
})
```

second local project config file :
```lua
local lsp = require('lsp-zero')

lsp.use('sumneko_lua',
{
	settings =
	{
		Lua =
		{
			runtime =
			{
				version = 'Lua 5.4'
			}
		}
	}
})
```

**Comment by VonHeikemen** [comment, marked as answer]
For that you would have to use `lspconfig` directly. `lsp-zero` will not let you change the configuration of a language server.

**Reply by ninofaivre** [reply]
Ok, thx have a good day ! :sunglasses:

**Comment by toondaey** [comment]
Hi! I'm not too familiar with lua but I was wondering how one can create the config in the project i.e. how do I name the config file.

**Reply by VonHeikemen** [reply]
If you are using neovim v0.9 you can enable the option `exrc`, like this.

```lua
vim.opt.exrc = true
```

And then you can create a config file called `.nvim.lua` in your project's folder.

If you are using Neovim v0.8 or lower you'll need to find a plugin that implements this feature.

**Reply by toondaey** [reply]
This works. Thanks!
